media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -243,7 +249,20 @@ bool MediaPipelineServerInternal::attachSourceInternal(const std::unique_ptr<Med
     }
 
     m_gstPlayer->attachSource(source);
-    source->setId(static_cast<int32_t>(source->getType()));
+
+    const auto kSourceIter = m_attachedSources.find(source->getType());
+    if (m_attachedSources.cend() == kSourceIter)
+    {
+        source->setId(generateSourceId());
```

Comment:
Maybe I'm missing something, but why don't we use the sourceId passed in by the client in the MediaSoruce object?

Reply:
Source ID is generated by Server, Client doesn't set this value in AttachSourceRequest, sourceId is a part of AttachSourceResponse:
message AttachSourceResponse {
    required int32 source_id = 1;
}